code/game/atoms.dm

```diff
@@ -57,9 +57,18 @@ var/global/list/ghdel_profiling = list()
 
 	var/arcanetampered = 0 //A looot of things can be
 
+	var/submerging = 0 // if atom is a liquid body for contents inside
+
 	var/image/moody_light
 	var/list/moody_lights = list()
 
+/atom/Entered(var/atom/movable/Obj, var/atom/OldLoc)
```

Comment:
This really only needs to apply to /obj rather than at the atom level (which includes mobs, obj, turfs, and areas); any reason why you want it at this level?

Reply:
oh i guess if people wanted turfs for people to be submerged in with this system

Reply:
Yeah save that for a separate feature PR, sounds like it's an easy change in the future.

Reply:
done that now